ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java

```diff
@@ -602,12 +603,23 @@ private QueryBlockInfo convertSource(RelNode r) throws CalciteSemanticException
 
       // retrieve the base table source.
       QueryBlockInfo tableFunctionSource = convertSource(tfs.getInput(0));
-      String sqAlias = tableFunctionSource.schema.get(0).table;
-      // the schema will contain the base table source fields
-      s = new Schema(tfs, sqAlias);
 
-      ast = createASTLateralView(tfs, s, tableFunctionSource, sqAlias);
+      // Create schema that preserves base table columns with original alias,
+      // but gives new UDTF columns the unique lateral view alias
+      int baseFieldCount = tableFunctionSource.schema.size();
+      List<RelDataTypeField> allOutputFields = tfs.getRowType().getFieldList();
 
+      final String sqAlias = tableFunctionSource.schema.get(0).table;
+      Stream<ColumnInfo> baseColumnsStream = allOutputFields.subList(0, baseFieldCount).stream()
+          .map(field -> new ColumnInfo(sqAlias, field.getName()));
+
+      final String lateralViewAlias = nextAlias();
+      Stream<ColumnInfo> udtfColumnsStream =
+          allOutputFields.subList(baseFieldCount, allOutputFields.size()).stream()
+              .map(field -> new ColumnInfo(lateralViewAlias, field.getName()));
+
+      s = new Schema(Stream.concat(baseColumnsStream, udtfColumnsStream).toList());
+      ast = createASTLateralView(tfs, s, tableFunctionSource, lateralViewAlias);
```

Comment:
Concretely, I was thinking something like the following:
```suggestion
      final String lateralViewAlias = nextAlias();
      ast = createASTLateralView(tfs, tableFunctionSource, lateralViewAlias);
      s = new Schema(tfs, lateralViewAlias);
```
assuming that the code inside `createASTLateralView` that needs a schema can use directly `tableFunctionSource.schema`.

In this case, the "input" schema is that inside `tableFunctionSource` and the "output" schema is the one constructed in `s`.